File: nodes/response-format/response-format.process.js
/**
 * Process function for the Response Format node.
 * Outputs a response_format object, either from the input or from the settings.
 */
export default async ({inputs, settings, config}) => {
  
  return {
    response_format: {
      type: "json_schema",
      json_schema: {
        name: inputs.name !== undefined ? inputs.name : settings.name,
        strict: inputs.strict !== undefined ? inputs.strict : settings.strict,
        schema: inputs.schema !== undefined ? inputs.schema : settings.schema
      }
    } 
  };
};

File: nodes/response-format/response-format.process.py
"""
Process function for the Response Format node.
Outputs a response_format object, either from the input or from the settings.
"""
from typing import Any, Dict

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    
    return {
        "response_format": {
            "type": "json_schema",
            "json_schema": {
                "name": inputs.get("name", settings.get("name")),
                "strict": inputs.get("strict", settings.get("strict")),
                "schema": inputs.get("schema", settings.get("schema"))
            }
        } 
    }
